src/main/java/Parser.java

```diff
@@ -0,0 +1,218 @@
+                            .trim().split(" /by ");
+                    String deadlineDescription = deadlineDescriptionDueDateArray[0];
+                    String deadlineDueDate = deadlineDescriptionDueDateArray[1];
+                    if (deadlineDescription.equals("")) { // empty description
+                        System.out.print("Missing task description!\n");
+                        System.out.print(dividerText);
+                    }
+                    else if (deadlineDueDate.equals("")) { // empty due date
+                        System.out.print("Missing task due date!\n");
+                        System.out.print(dividerText);
+                    }
+                    else if (deadlineDescription.contains("/from") ||
+                            deadlineDescription.contains("/to")) { // unnecessary from and to date
+                        System.out.print("Deadline task cannot have a from and to date!\n");
+                        System.out.print(dividerText);
+                    }
+                    else {
+                        try {
+                            LocalDate deadlineDueDateLocal = LocalDate.parse(deadlineDueDate);
+                            existingTaskList.addDeadline(deadlineDescription, deadlineDueDateLocal);
+                        }
+                        catch (Exception e) {
+                            System.out.print("Incorrect formatting of due date! Ensure it is yyyy-mm-dd\n");
+                            System.out.print(dividerText);
+                        }
+                    }
+                    Storage.saveTaskList(existingTaskList);
+                    return 1;
+
+                case "event":
+                    String eventDescriptionFromTo = input.replace("event", "").trim();
+                    if (!eventDescriptionFromTo.contains(" /from ") ||
+                            !eventDescriptionFromTo.contains(" /to ")) { // missing '/from' or '/to' keywords
+                        System.out.print("Missing event from and/or to date!\n");
+                        System.out.print(dividerText);
+                        Storage.saveTaskList(existingTaskList);
+                        return 1;
+                    }
+                    String[] eventDescriptionFromToArray = input.replace("event", "")
+                            .trim().split(" /from ");
+                    String eventDescription = eventDescriptionFromToArray[0];
+                    String eventFromTo = eventDescriptionFromToArray[1];
+                    String[] eventFromToArray;
+                    String eventFrom = "";
+                    String eventTo = "";
+                    if (eventFromTo.contains(" /to ")) {
+                        eventFromToArray = eventDescriptionFromToArray[1].trim().split(" /to ");
+                        eventFrom = eventFromToArray[0];
+                        eventTo = eventFromToArray[1];
+                    }
```

Comment:
Can consider leaving a blank line after this line of code to enhance the readability.

"Logical units within a block should be separated by one blank line." - Java coding standard website.